source/common/upstream/subset_lb.h

```diff
@@ -262,6 +262,7 @@ class SubsetLoadBalancer : public LoadBalancer, Logger::Loggable<Logger::Id::ups
 
   std::string single_key_;
   absl::flat_hash_map<HashedValue, HostConstSharedPtr> single_host_per_subset_map_;
+  Stats::Gauge* single_duplicate_stat_{};
```

Comment:
i assume this member variable doesn't need locking, as it would be instantiated per worker thread. Is that right?

Reply:
Yes. I think only "thread aware" load balancers need to think about threading/locking; this state is all copied per-thread.